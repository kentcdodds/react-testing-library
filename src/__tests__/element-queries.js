import React from 'react'
import {render} from '../'

const TestComponent = () => <span data-testid="test-component" />

test('queryByTestId finds matching element', () => {
  const {queryByTestId} = render(<TestComponent />)
  expect(queryByTestId('test-component')).toMatchSnapshot()
})

test('queryByTestId returns null when no matching element exists', () => {
  const {queryByTestId} = render(<TestComponent />)
  expect(queryByTestId('unknown-data-testid')).toBeNull()
})

test('getByTestId finds matching element', () => {
  const {getByTestId} = render(<TestComponent />)
  expect(getByTestId('test-component')).toMatchSnapshot()
})

test('getByTestId throws error when no matching element exists', () => {
  const {getByTestId} = render(<TestComponent />)
  expect(() =>
    getByTestId('unknown-data-testid'),
  ).toThrowErrorMatchingSnapshot()
})
