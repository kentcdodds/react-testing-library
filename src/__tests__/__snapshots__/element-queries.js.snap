// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getByTestId finds matching element 1`] = `
<span
  data-testid="test-component"
/>
`;

exports[`getByTestId throws error when no matching element exists 1`] = `"Unable to find element by [data-testid=\\"unknown-data-testid\\"]"`;

exports[`queryByTestId finds matching element 1`] = `
<span
  data-testid="test-component"
/>
`;
